### PR TITLE
MAINT: Apply ruff/flake8-raise rules (RSE)

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
In preparation for https://github.com/numpy/numpy/issues/24994.

### CODE REVIEW

File: numpy/polynomial/chebyshev.py [hunk 1]
```diff
@@ -800,7 +800,7 @@ def chebdiv(c1, c2):
     # c1, c2 are trimmed copies
     [c1, c2] = pu.as_series([c1, c2])
     if c2[-1] == 0:
-        raise ZeroDivisionError()
+        raise ZeroDivisionError
```
Comment (mhvk) [inline numpy/polynomial/chebyshev.py:803]: The trouble with automated fixes is that one sees things that really should be better, such as here: why is there no helpful message? I guess one can hope to find this again, but seems a bit sad.

File: tools/c_coverage/c_coverage_report.py [hunk 2]
```diff
@@ -12,7 +12,7 @@
 try:
     import pygments
     if tuple([int(x) for x in pygments.__version__.split('.')]) < (0, 11):
-        raise ImportError()
+        raise ImportError
```
Comment (mhvk) [inline tools/c_coverage/c_coverage_report.py:15]: Similarly, here, but probably rarely seen and obvious from context.

**Review by mhvk — APPROVED** [review]
Also looks good to me, though given my comments I won't immediately merge -- need to decide a bit whether we are just "following some rules" or trying to actually make the code better in the process...

**Comment by DimitriPapadopoulos** [comment]
This can be left for later, after migrating to ruff.

File: numpy/polynomial/chebyshev.py (hunk 1 above)
Comment (charris) [reply numpy/polynomial/chebyshev.py:803]: Could add a `FIXME` comment.

Comment (DimitriPapadopoulos) [reply numpy/polynomial/chebyshev.py:803]: Good idea, I'll add FIXME comments.

**Comment by charris** [comment]
Thanks @DimitriPapadopoulos .

**Pull request merged**